### PR TITLE
Issue98 simplify smart contract service adjustments

### PR DESCRIPTION
**Pull request by leeduan**
Refactor service for smart contract changes:

* Remove governance endpoints (join, accept)
* Handle smart contract event attribute adjustments
* Remove some unused dependencies/upgrade couple libs
* RedeemBurn -> Burn since no longer need to deal with bank coin
* Remove bank settlement and transfer to another member bank functionality not needed

### CODE REVIEW

**Review by ranadewall — APPROVED** [review]

File: service/src/main/kotlin/io/provenance/digitalcurrency/consortium/domain/CoinBurn.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+import org.jetbrains.exposed.dao.id.EntityID
+import org.jetbrains.exposed.sql.and
+import java.util.UUID
+
+typealias CBT = CoinBurnTable
+
+object CoinBurnTable : BaseCoinRequestTable(name = "coin_burn_v2")
+
+open class CoinRedeemBurnEntityClass : BaseCoinRequestEntityClass<CBT, CoinBurnRecord>(CBT) {
```
Comment (ranadewall) [inline service/src/main/kotlin/io/provenance/digitalcurrency/consortium/domain/CoinBurn.kt:11]: naming - `CoinBurnEntityClass`

File: service/src/main/kotlin/io/provenance/digitalcurrency/consortium/frameworks/BurnCompletionQueue.kt [hunk 2]
```diff
@@ -27,7 +27,7 @@ class RedeemBurnCompletionQueue(
     private val bankClient: BankClient,
```
Comment (ranadewall) [inline service/src/main/kotlin/io/provenance/digitalcurrency/consortium/frameworks/BurnCompletionQueue.kt:27]: class needs to be renamed

Comment (ranadewall) [reply service/src/main/kotlin/io/provenance/digitalcurrency/consortium/frameworks/BurnCompletionQueue.kt:27]: nit - and the log message in `startProcessing`

**Comment by ranadewall** [comment]
there's still a redeem wrapper in eventstreamconsumer - not sure if we are leaving this for catching existing events and it should be deprecated, or if it should go away.

File: service/src/main/kotlin/io/provenance/digitalcurrency/consortium/domain/CoinBurn.kt (hunk 1 above)
Comment (leeduan) [reply service/src/main/kotlin/io/provenance/digitalcurrency/consortium/domain/CoinBurn.kt:11]: doh

File: service/src/main/kotlin/io/provenance/digitalcurrency/consortium/service/BankService.kt [hunk 3]
```diff
@@ -120,9 +113,6 @@ class BankService(
                         check(registration.isActive()) { "Cannot transfer to removed bank account $bankAccountUuid" }
                         CoinTransferRecord.insert(uuid, registration, amount)
                     }
-                    // Sending to another member bank
-                    blockchainAddress != null && pbcService.getMembers().members.any { it.id == blockchainAddress } ->
```
Comment (ranadewall) [inline service/src/main/kotlin/io/provenance/digitalcurrency/consortium/service/BankService.kt:124]: we're still validating the blockchain address but we aren't actually using it to do anything with this bit removed.

Comment (leeduan) [reply service/src/main/kotlin/io/provenance/digitalcurrency/consortium/service/BankService.kt:124]: sorry not sure i follow, any reason to keep this?

Comment (ranadewall) [reply service/src/main/kotlin/io/provenance/digitalcurrency/consortium/service/BankService.kt:124]: if we're going to allow a transfer request to a block chain address then yes. If not then we should take the blockchain address off the `TransferRequest` object.

**Pull request merged**